src/main/java/bio/terra/service/filedata/azure/tables/TableDao.java

```diff
@@ -164,6 +165,27 @@ public FSItem retrieveByPath(
         fullPath);
   }
 
+  public Optional<FSItem> lookupOptionalPath(
+      UUID datasetId, String fullPath, AzureStorageAuthInfo storageAuthInfo, int enumerateDepth) {
+    TableServiceClient tableServiceClient =
+        azureAuthService.getTableServiceClient(
+            storageAuthInfo.getSubscriptionId(),
+            storageAuthInfo.getResourceGroupName(),
+            storageAuthInfo.getStorageAccountResourceName());
```

Comment:
It looks like this usage is very common in our code base. Adding an overload `AzureAuthService.getTableServiceClient(AzureStorageAuthInfo)` would let us delete a lot of duplicate code.
```suggestion
        azureAuthService.getTableServiceClient(storageAuthInfo);
```

Reply:
This is something I've been wanting to do, but want to do it in a separate PR as to not pollute feature PRs

Reply:
https://broadworkbench.atlassian.net/browse/DR-2157

Reply:
https://github.com/DataBiosphere/jade-data-repo/pull/1082